### PR TITLE
Add support of *.xml files #3

### PR DESCRIPTION
**Pull request by kylec32**
If both are available uses JSON. Also allows specifying file extension to override default ordering.

### CODE REVIEW

**Comment by kylec32** [comment]
Cleaning up those errors now.

File: tests/test_utils.py [hunk 1]
```diff
@@ -10,6 +10,28 @@ def test_pull():
     assert result
 
 
+def test_pullPriority():
```
Comment (lk-geimfari) [inline tests/test_utils.py]: Please, look at PEP8. The names of variables and functions should be:

`name_of_function()`, `name_of_var`.

File: wasserstoff/utils.py [hunk 2]
```diff
@@ -1,19 +1,59 @@
 from collections import Mapping
 import json
+import os.path
+import xml.etree.ElementTree as ElementTree
+from wasserstoff.xmlHelpers import (
+    XmlDictConfig,
+)
 
 
 def pull(filename):
```
Comment (lk-geimfari) [inline wasserstoff/utils.py]: just add where parameter `fmt` which will use for detecting file extension.

File: wasserstoff/utils.py [hunk 3]
```diff
@@ -1,19 +1,59 @@
...
     :rtype: dict
     """
-    with open(filename + '.json', 'r') as file:
+    if fileExists(filename, 'json') and fileExtension(filename) != 'xml':
+        return pullJson(filename)
+    else:
+        return pullXml(filename)
+
+
+def pullXml(filename):
```
Comment (lk-geimfari) [inline wasserstoff/utils.py]: PEP8: https://www.python.org/dev/peps/pep-0008/#function-names

File: wasserstoff/utils.py [hunk 4]
```diff
@@ -1,19 +1,59 @@
...
+    tree = ElementTree.parse(addExtensionIfNotThere(filename, 'xml'))
+    root = tree.getroot()
+    xmldict = XmlDictConfig(root)
+    return xmldict
+
+
+def pullJson(filename):
```
Comment (lk-geimfari) [inline wasserstoff/utils.py]: PEP8

File: wasserstoff/utils.py [hunk 5]
```diff
@@ -1,19 +1,59 @@
...
+    with open(addExtensionIfNotThere(filename, 'json'), 'r') as file:
         data = json.load(file)
     return data
 
 
+def addExtensionIfNotThere(filename, extension):
```
Comment (lk-geimfari) [inline wasserstoff/utils.py]: pep8

File: wasserstoff/utils.py [hunk 6]
```diff
@@ -1,19 +1,59 @@
...
+    if fileExtension(filename) != extension:
+        return filename + '.' + extension
+    else:
+        return filename
+
+
+def fileExists(filename, extension):
```
Comment (lk-geimfari) [inline wasserstoff/utils.py]: here too

**Comment by lk-geimfari** [comment]
@kylec32 I recommend you to read  [PEP8](https://www.python.org/dev/peps/pep-0008/).. This is a really nice style guide.

File: README.rst [hunk 7]
```diff
@@ -58,6 +58,17 @@ Content of ``test.json``:
       "secret_key": "001110110100101100101010100010111010"
     }
 
+alternatively
+
+Content of ``test.xml``:
+
+.. code:: xml
+
+    {
```
Comment (lk-geimfari) [inline README.rst]: This is not xml, it's json.

**Comment by lk-geimfari** [comment]
@kylec32 Also, unfortunately, i don't see stylizing of `key` from `*xml`.

**Pull request closed without merging**